test/Microsoft.ML.TimeSeries.Tests/TimeSeries.cs

```diff
@@ -124,6 +124,8 @@ public void SavePipeSlidingWindowW1L1()
         }
 
         [Fact]
+        //Skipping test temporarily. This test will be re-enabled once the cause of failures has been determined
+        [Trait("Category", "SkipInCI")]
         public void SavePipeSlidingWindowW2L1()
```

Comment:
This test is from 2016.

Seems pretty odd that it would suddenly turn flaky. Is there some common cause for the tests being unstable? I fully anticipate new ones like the TF tests would be suspect (as they are less battle hardened), but old tests, I would expect to be pretty stable.

Reply:
Thanks Justin, we are not fully understand all the flakyness now but we will investigate one by one.

---
In reply to: [372261206](https://github.com/dotnet/machinelearning/pull/4726#discussion_r372261206) [](ancestors = 372261206)

Reply:
@justinormont I do wonder: you said this tests exists since 2016, and it wasn't flaky. Was it the case that you had to run the tests on the Azure CI Pipelines?

On my computer pretty much all the tests pass on the first run, and it's only on Azure pipelines that they become flaky, that's why I am asking.

Reply:
>  I do wonder: you said this tests exists since 2016, and it wasn't flaky. Was it the case that you had to run the tests on the Azure CI Pipelines?

This test was running on the Azure CI Pipelines since the ML.NET repo started using it. As the 2016 date implies, the unit test pre-exists the ML.NET repo, and originates in TLC.

> On my computer pretty much all the tests pass on the first run, and it's only on Azure pipelines that they become flaky, that's why I am asking.

Is there something interesting about the Azure CI Pipelines which makes unit tests unstable, yet stable in local runs? 

E.g. build server clears the build cache between builds/tests (starting from a clean slate); build server runs everything in parallel; has 32-cores vs. local machine has 4-cores; some oddity causing all unit tests to run in the same process but in a different order (causing different random numbers to be generated for each segment of the run); or odd math differences like one environment having gcc's `--ffast-math` enabled/disabled (`/fp:precise`  in VS) causing small round-off errors to accumulate.

Reply:
@frank-dong-ms has mentioned to me that some tests pass when running them in isolation on the pipelines, but they tend to be flaky when ran along with all the other tests. So it seems that tests do have an effect on each other.

About the other things you mention, it might also be worth of investigation.